cgo/extern.go

```diff
@@ -69,6 +69,7 @@ func cgo_extern_verify_consensus_fault(
 	miner_id *C.uint64_t,
 	epoch *C.int64_t,
 	fault *C.int64_t,
+	gas_used *C.int64_t,
```

Comment:
Given that we're already using an "out" pointer here, what if we passed in a pointer to some `gas_remaining` variable? That would let us stop early without changing the parameters. I.e.:

1. Read the current gas remaining from the `gas_remaining` parameter.
2. Update it locally, stopping if/when we run out.
3. Write back the final value.

Reply:
Although I guess this might complicate some other parts of the API a bit?